### PR TITLE
Feature/sg 747 Adding utils to load/save videos

### PR DESCRIPTION
**Pull request by Louis-Dupont**
This is the first part of the feature to support videos

### CODE REVIEW

**Review by shaydeci — DISMISSED** [review]
LGTM

File: src/super_gradients/training/utils/videos.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    for frame in frames:
+        if frame.ndim == 2:
+            frame = frame[:, :, np.newaxis]
+        video_writer.write(frame)
+
+    video_writer.release()
+
+
+def _validate_frames(frames: List[np.ndarray]) -> Tuple[float, float]:
```
Comment (BloodAxe) [inline src/super_gradients/training/utils/videos.py:84]: Why do we need this function? One can easily stack frames to [B,H,W,C] that would guarantee size constraint

Comment (shaydeci) [reply src/super_gradients/training/utils/videos.py:84]: That won't work if H,W are not the same...

File: src/super_gradients/training/utils/videos.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import numpy as np
+
+SUPPORTED_FORMATS = (".avi", ".mp4", ".mov", ".wmv", ".mkv")
+
+
+__all__ = ["load_video", "save_video"]
+
+
+def load_video(file_path: str, max_frames: Optional[int] = None) -> List[np.ndarray]:
+    """Open a video file and extract each frame into numpy array.
```
Comment (BloodAxe) [inline src/super_gradients/training/utils/videos.py:14]: OpenCV loads images as BGR. I believe this should be reflected in documentation if it's intended. Or BGR->RGB conversion should be done inside. Save for image saving. When saving
OpenCV assumes input image is in BGR order.

File: src/super_gradients/training/utils/videos.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def _open_video(file_path: str) -> cv2.VideoCapture:
+    """Open a video file.
+
+    :param file_path:   Path to the video file
+    :return:            Opened video capture object
+    """
+    ext = os.path.splitext(file_path)[-1].lower()
+    if ext not in SUPPORTED_FORMATS:
```
Comment (BloodAxe) [inline src/super_gradients/training/utils/videos.py:33]: Well, supported formats is defined inside OpenCV library and why not we rely on `if not cap.isOpened()` only? This check looks unnecessary to me.

File: src/super_gradients/training/utils/videos.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/utils/videos.py:84]: It doesnt work if H,W are not the same, we could pad but I don't think this should be our responsability, this should be done before

File: src/super_gradients/training/utils/videos.py (hunk 3 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/utils/videos.py:33]: I wasnt aware of that, thanks for that info

File: src/super_gradients/training/utils/videos.py (hunk 2 above)
Comment (Louis-Dupont) [reply src/super_gradients/training/utils/videos.py:11]: Fixed in d2be71775d822f2a304b9f73044118add2ecaefd

**Review by BloodAxe — APPROVED** [review]
LGTM

**Pull request merged**